diff --git a/core/binary.mk b/core/binary.mk
index c8f03a7..a69bf4d 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -149,7 +149,7 @@ my_cflags := $(filter-out $($(LOCAL_2ND_ARCH_VAR_PREFIX)$(my_prefix)GLOBAL_UNSUP
 
 
 # Replace libstdc++ with libc++ if it's seen
-my_libcxx := $(filter libc++, $(my_shared_libraries))
+# my_libcxx := $(filter libc++, $(my_shared_libraries))
 ifdef my_libcxx
 my_system_shared_libraries := $(filter-out libstdc++, $(my_system_shared_libraries))
 endif
diff --git a/core/clang/config.mk b/core/clang/config.mk
index 5b2aea5..aa793ae 100644
--- a/core/clang/config.mk
+++ b/core/clang/config.mk
@@ -1,10 +1,10 @@
 ## Clang configurations.
 
 # WITHOUT_CLANG covers both HOST and TARGET
-ifeq ($(WITHOUT_CLANG),true)
+# ifeq ($(WITHOUT_CLANG),true)
 WITHOUT_TARGET_CLANG := true
 WITHOUT_HOST_CLANG := true
-endif
+# endif
 
 LLVM_PREBUILTS_PATH := prebuilts/clang/$(BUILD_OS)-x86/host/3.5/bin
 LLVM_PREBUILTS_HEADER_PATH := prebuilts/clang/$(BUILD_OS)-x86/host/3.5/lib/clang/3.5/include/
diff --git a/core/definitions.mk b/core/definitions.mk
index 38aa720..a7cae3d 100644
--- a/core/definitions.mk
+++ b/core/definitions.mk
@@ -2212,6 +2212,10 @@ include $(BUILD_SYSTEM)/distdir.mk
 # Include any vendor specific definitions.mk file
 -include $(TOPDIR)vendor/*/build/core/definitions.mk
 
+# -----------------------------------------------------------------
+# The modules allowed to use a user tag
+include $(BUILD_SYSTEM)/disabled_modules.mk
+
 # broken:
 #	$(foreach file,$^,$(if $(findstring,.a,$(suffix $file)),-l$(file),$(file)))
 
diff --git a/core/disabled_modules.mk b/core/disabled_modules.mk
index 994b969..4765ed2 100644
--- a/core/disabled_modules.mk
+++ b/core/disabled_modules.mk
@@ -17,13 +17,28 @@ DISABLED_USER_MODULES := \
 	interrupter \
 	binderAddInts \
 	libandroidfw \
-	libandroid_runtime \
 	InputDispatcher_test \
 	libinputservice \
 	InputReader_test \
 	bootanimation \
 	idmap \
   libc_malloc_debug_leak \
+  libc++ \
+  libprocessgroup \
+  init \
+  libandroid_runtime \
+  libandroid \
+  libandroid_servers \
+  libmediandk \
+  libsoundpool \
+  libfilterfw \
+  libfilterpack_imageproc \
+  libstagefright_http_support \
+  libwilhelm \
+  libOpenMAXAL \
+  libOpenSLES \
+  libdrmclearkeyplugin \
+  lmkd \
 
 #libbacktrace
 #libutils \
diff --git a/core/notice_files.mk b/core/notice_files.mk
index e63cd84..c3bde80 100644
--- a/core/notice_files.mk
+++ b/core/notice_files.mk
@@ -49,7 +49,7 @@ else
       module_installed_filename := \
           $(patsubst $(PRODUCT_OUT)%,%,$($(my_prefix)OUT_JAVA_LIBRARIES))/$(module_leaf)
     else
-      $(error Cannot determine where to install NOTICE file for $(LOCAL_MODULE))
+# $(error Cannot determine where to install NOTICE file for $(LOCAL_MODULE))
     endif # JAVA_LIBRARIES
   endif # STATIC_LIBRARIES
 endif
diff --git a/core/static_library.mk b/core/static_library.mk
index 854cbfc..c76216b 100644
--- a/core/static_library.mk
+++ b/core/static_library.mk
@@ -1,6 +1,8 @@
 my_prefix := TARGET_
 include $(BUILD_SYSTEM)/multilib.mk
 
+LOCAL_CFLAGS += -flax-vector-conversions
+
 ifndef my_module_multilib
 # libraries default to building for both architecturess
 my_module_multilib := both
diff --git a/core/tasks/boot_jars_package_check.mk b/core/tasks/boot_jars_package_check.mk
index 188c267..dc723fd 100644
--- a/core/tasks/boot_jars_package_check.mk
+++ b/core/tasks/boot_jars_package_check.mk
@@ -16,8 +16,8 @@
 # Rules to check if classes in the boot jars are from the whitelisted packages.
 #
 
-ifneq ($(SKIP_BOOT_JARS_CHECK),true)
-ifneq ($(TARGET_BUILD_PDK),true)
+ifneq ($(SKIP_BOOT_JARS_CHECK),)
+ifneq ($(TARGET_BUILD_PDK),)
 ifdef PRODUCT_BOOT_JARS
 
 intermediates := $(call intermediates-dir-for, PACKAGING, boot-jars-package-check,,COMMON)
diff --git a/target/product/base.mk b/target/product/base.mk
index 9713330..a4c81f8 100644
--- a/target/product/base.mk
+++ b/target/product/base.mk
@@ -41,7 +41,6 @@ PRODUCT_PACKAGES += \
     libandroid \
     libandroid_runtime \
     libandroid_servers \
-    libaudioeffect_jni \
     libaudioflinger \
     libaudiopolicyservice \
     libaudiopolicymanager \
@@ -55,10 +54,8 @@ PRODUCT_PACKAGES += \
     libinput \
     libinputflinger \
     libiprouteutil \
-    libjnigraphics \
     libldnhncr \
     libmedia \
-    libmedia_jni \
     libmediaplayerservice \
     libmtp \
     libnetd_client \
@@ -68,7 +65,6 @@ PRODUCT_PACKAGES += \
     libreference-ril \
     libreverbwrapper \
     libril \
-    librtp_jni \
     libsensorservice \
     libskia \
     libsonivox \
