From 57fc9e77f8407a920857b2807451a1168a586f19 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 7 Apr 2015 19:06:24 +0800
Subject: [PATCH] Bug 1151787 - Disable MOZJEMALLOC in gecko if bionic defined
 it. r=mwu

---
 Android.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Android.mk b/Android.mk
index 68b39f4..f1074e7 100644
--- a/Android.mk
+++ b/Android.mk
@@ -173,6 +173,12 @@ ifeq ($(B2G_SYSTEM_APPS),1)
 PRESERVE_B2G_WEBAPPS := 1
 endif
 
+ifeq ($(PLATFORM_SDK_VERSION),22)
+ifneq ($(MALLOC_IMPL),dlmalloc)
+DISABLE_JEMALLOC := 1
+endif
+endif
+
 ifeq ($(ENABLE_GLOBAL_PRELINK),1)
 APRIORI := $(HOST_OUT_EXECUTABLES)/apriori$(HOST_EXECUTABLE_SUFFIX)
 PRELINK_MAP := $(abspath $(LOCAL_PATH)/prelink.map)
