diff --git a/cmds/interrupter/Android.mk b/cmds/interrupter/Android.mk
index e324627..b3744da 100644
--- a/cmds/interrupter/Android.mk
+++ b/cmds/interrupter/Android.mk
@@ -16,6 +16,6 @@ LOCAL_SRC_FILES := \
     interrupter.c
 LOCAL_MODULE := interrupter
 LOCAL_MODULE_TAGS := eng tests
-LOCAL_LDFLAGS := -ldl
+LOCAL_LDFLAGS := -Wl,--no-as-needed -ldl
 
-include $(BUILD_HOST_SHARED_LIBRARY)
\ No newline at end of file
+include $(BUILD_HOST_SHARED_LIBRARY)
