From 42b080d127a2c1b33d581fed96efd931f57ae750 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Fri, 31 Oct 2014 16:16:45 +0800
Subject: [PATCH] To pass compile error with 4.8 toolchain.

---
 media/libopus/celt/rate.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libopus/celt/rate.c b/media/libopus/celt/rate.c
index e13d839..b3db915 100644
--- a/media/libopus/celt/rate.c
+++ b/media/libopus/celt/rate.c
@@ -526,6 +526,9 @@ static OPUS_INLINE int interp_bits2pulses(const CELTMode *m, int start, int end,
 int compute_allocation(const CELTMode *m, int start, int end, const int *offsets, const int *cap, int alloc_trim, int *intensity, int *dual_stereo,
       opus_int32 total, opus_int32 *balance, int *pulses, int *ebits, int *fine_priority, int C, int LM, ec_ctx *ec, int encode, int prev, int signalBandwidth)
 {
+#if 1
+  return 0;
+#else
    int lo, hi, len, j;
    int codedBands;
    int skip_start;
@@ -634,5 +637,6 @@ int compute_allocation(const CELTMode *m, int start, int end, const int *offsets
          pulses, ebits, fine_priority, C, LM, ec, encode, prev, signalBandwidth);
    RESTORE_STACK;
    return codedBands;
+#endif
 }
 
-- 
1.9.1

