From 113295e2b40752845f1ad0fc0e019e31730a3bfc Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Wed, 29 Oct 2014 16:52:10 +0800
Subject: [PATCH 01/13] Add api level 20.

---
 b2g/app/moz.build |  2 +-
 configure.in      | 15 +++++++++++++++
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/b2g/app/moz.build b/b2g/app/moz.build
index 948e70a..c871a86 100644
--- a/b2g/app/moz.build
+++ b/b2g/app/moz.build
@@ -60,7 +60,7 @@ if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gonk':
         'cutils',
     ]
     OS_LIBS += CONFIG['MOZ_ZLIB_LIBS']
-    if CONFIG['ANDROID_VERSION'] in ('17', '18', '19'):
+    if CONFIG['ANDROID_VERSION'] in ('17', '18', '19', '21'):
         OS_LIBS += [
             'gui',
             'suspend',
diff --git a/configure.in b/configure.in
index 191d8be..3f7e02a 100644
--- a/configure.in
+++ b/configure.in
@@ -287,6 +287,21 @@ if test -n "$gonkdir" ; then
         AC_DEFINE(MOZ_AUDIO_OFFLOAD)
 	MOZ_FMP4=1
         ;;
+    21)
+        GONK_INCLUDES="-I$gonkdir/frameworks/native/include -I$gonkdir/frameworks/av/include -I$gonkdir/frameworks/av/include/media -I$gonkdir/frameworks/av/include/camera -I$gonkdir/frameworks/native/include/media/openmax -I$gonkdir/frameworks/av/media/libstagefright/include"
+#MOZ_B2G_CAMERA=1
+#MOZ_B2G_BT=1
+#MOZ_B2G_BT_BLUEDROID=1
+#MOZ_NFC=1
+#MOZ_RTSP=1
+#MOZ_OMX_DECODER=1
+#MOZ_OMX_ENCODER=1
+#AC_DEFINE(MOZ_OMX_ENCODER)
+#MOZ_AUDIO_OFFLOAD=1
+#AC_SUBST(MOZ_AUDIO_OFFLOAD)
+#AC_DEFINE(MOZ_AUDIO_OFFLOAD)
+#MOZ_FMP4=1
+        ;;
     *)
         AC_MSG_ERROR([Unsupported platform version: $ANDROID_VERSION])
         ;;
-- 
1.9.1

