From 10eacef696a0dcb93eb498bf725272037f35d3c8 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 28 Oct 2014 23:04:36 +0800
Subject: [PATCH] Pass BionicGlue compile error.

---
 mozglue/build/BionicGlue.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mozglue/build/BionicGlue.cpp b/mozglue/build/BionicGlue.cpp
index ec4ec6a..1bdd3fa 100644
--- a/mozglue/build/BionicGlue.cpp
+++ b/mozglue/build/BionicGlue.cpp
@@ -169,7 +169,7 @@ __wrap_PR_SetEnv(const char *string)
     if ( !strchr(string, '=')) return(-1);
 
     pthread_mutex_lock(&_pr_envLock);
-    result = putenv(string);
+    result = putenv((char*)string);
     pthread_mutex_unlock(&_pr_envLock);
     return (result)? -1 : 0;
 }
-- 
1.9.1

