From 3ef29789481f94e933aa70ea4b0a84ac3b44b2be Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Sun, 16 Nov 2014 23:20:14 +0800
Subject: [PATCH 03/14] Update build/gyp.mozbuild for gonk-L.

---
 build/gyp.mozbuild | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gyp.mozbuild b/build/gyp.mozbuild
index 950804a..41dbbc6 100644
--- a/build/gyp.mozbuild
+++ b/build/gyp.mozbuild
@@ -73,7 +73,7 @@ elif os == 'Android':
         gyp_vars['build_with_gonk'] = 1
         gyp_vars['moz_widget_toolkit_gonk'] = 1
         gyp_vars['opus_complexity'] = 1
-        if int(CONFIG['ANDROID_VERSION']) >= 18:
+        if CONFIG['ANDROID_VERSION'] in ('18', '19'):
           gyp_vars['moz_webrtc_omx'] = 1
     else:
         gyp_vars.update(
-- 
1.9.1

