From 90b2d5ceed5cb7a5c6bc186f2f7995275a85fa0b Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Sun, 16 Nov 2014 23:20:46 +0800
Subject: [PATCH 05/14] Update dom/system/gonk/systemlibs.js for gonk-L.

---
 dom/system/gonk/systemlibs.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dom/system/gonk/systemlibs.js b/dom/system/gonk/systemlibs.js
index b76822f..1435396 100644
--- a/dom/system/gonk/systemlibs.js
+++ b/dom/system/gonk/systemlibs.js
@@ -129,6 +129,8 @@ this.libnetutils = (function() {
     ifc_disable: library.declare("ifc_disable", ctypes.default_abi,
                                  ctypes.int,
                                  ctypes.char.ptr),
+    /* this is removed in L release
+     * need to use ifc_add_route instead
     ifc_add_host_route: library.declare("ifc_add_host_route",
                                         ctypes.default_abi,
                                         ctypes.int,
@@ -147,6 +149,7 @@ this.libnetutils = (function() {
                                            ctypes.default_abi,
                                            ctypes.int,
                                            ctypes.char.ptr),
+    */
     ifc_remove_default_route: library.declare("ifc_remove_default_route",
                                               ctypes.default_abi,
                                               ctypes.int,
@@ -159,6 +162,7 @@ this.libnetutils = (function() {
                                    ctypes.int,
                                    ctypes.int,
                                    ctypes.int),
+    /*
     ifc_add_route: library.declare("ifc_add_route", ctypes.default_abi,
                                    ctypes.int, // return value
                                    ctypes.char.ptr, // ifname
@@ -171,6 +175,7 @@ this.libnetutils = (function() {
                                       ctypes.char.ptr, // dst
                                       ctypes.int, // prefix_length
                                       ctypes.char.ptr), // gw
+    */
     dhcp_stop: library.declare("dhcp_stop", ctypes.default_abi,
                                ctypes.int,
                                ctypes.char.ptr),
-- 
1.9.1

