From 026eecc9541affb1de7d4540d2fd6b56bfc56dea Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Wed, 29 Oct 2014 16:57:43 +0800
Subject: [PATCH 07/13] Fix dom/icc compile error.

---
 dom/icc/Icc.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dom/icc/Icc.h b/dom/icc/Icc.h
index 37cfd57..3f89003 100644
--- a/dom/icc/Icc.h
+++ b/dom/icc/Icc.h
@@ -119,6 +119,7 @@ public:
   IMPL_EVENT_HANDLER(stksessionend)
 
 private:
+  ~Icc() {}
   bool mLive;
   uint32_t mClientId;
   nsString mIccId;
-- 
1.9.1

