From f1bee84470ae870a409c167ebb1cb40b8b6a6c63 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Sun, 16 Nov 2014 23:22:31 +0800
Subject: [PATCH 11/14] Update security/nss/lib/freebl/unix_rand.c for gonk-L.

---
 security/nss/lib/freebl/unix_rand.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/security/nss/lib/freebl/unix_rand.c b/security/nss/lib/freebl/unix_rand.c
index 579040ee..99477a0 100644
--- a/security/nss/lib/freebl/unix_rand.c
+++ b/security/nss/lib/freebl/unix_rand.c
@@ -735,7 +735,14 @@ safe_popen(char *cmd)
 	 */
 	if (!freopen("/dev/null", "r", stdin))
 	    close(0);
+#if defined(HAVE_SYSCONF) && defined(_SC_OPEN_MAX)
+	ndesc = sysconf(_SC_OPEN_MAX);
+#elif defined(HAVE_GETDTABLESIZE)
 	ndesc = getdtablesize();
+#else
+  //#warning Using default value
+  ndesc = 123;
+#endif
 	for (fd = PR_MIN(65536, ndesc); --fd > 2; close(fd));
 
 	/* clean up environment in the child process */
-- 
1.9.1

