From 493d4c3c04377fada16f6891a4d36651cf780bb0 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Sun, 16 Nov 2014 23:22:58 +0800
Subject: [PATCH 13/14] Update tools/ for gonk-L.

---
 tools/profiler/moz.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/profiler/moz.build b/tools/profiler/moz.build
index f2133bc..89da893 100644
--- a/tools/profiler/moz.build
+++ b/tools/profiler/moz.build
@@ -100,7 +100,7 @@ if CONFIG['MOZ_ENABLE_PROFILER_SPS']:
     if CONFIG['ENABLE_TESTS']:
         DIRS += ['tests/gtest']
 
-    if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gonk' and CONFIG['ANDROID_VERSION'] <= '17':
+    if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gonk' and (CONFIG['ANDROID_VERSION'] <= '17' or CONFIG['ANDROID_VERSION'] >= '21'):
         DEFINES['ELFSIZE'] = 32
 
     FINAL_LIBRARY = 'xul'
-- 
1.9.1

