diff --git a/media/libopus/moz.build b/media/libopus/moz.build
index 2607a19..55ddc40 100644
--- a/media/libopus/moz.build
+++ b/media/libopus/moz.build
@@ -88,14 +88,23 @@ if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['GNU_AS']:
     GENERATED_SOURCES += [ '%s.%s' % (f, CONFIG['ASM_SUFFIX']) for f in [
         'celt_pitch_xcorr_arm-gnu',
     ]]
-    # -Os is significantly slower, enable -O3 unless optimization is disabled
     if CONFIG['MOZ_OPTIMIZE']:
-      CFLAGS += [
-        '-O3',
-      ]
-      CXXFLAGS += [
-        '-O3',
-      ]
+      if CONFIG['ANDROID_VERSION'] >= '21':
+        # -O3 hit gcc bug in gonk-l, fallback to use -Os
+        CFLAGS += [
+          '-Os',
+        ]
+        CXXFLAGS += [
+          '-Os',
+        ]
+      else:
+        # -Os is significantly slower, enable -O3 unless optimization is disabled
+        CFLAGS += [
+          '-O3',
+        ]
+        CXXFLAGS += [
+          '-O3',
+        ]
 
 # Suppress warnings in third-party code.
 if CONFIG['GNU_CC']:
