diff --git a/configure.in b/configure.in
index 81a44ea..ec8f4b0 100644
--- a/configure.in
+++ b/configure.in
@@ -3415,7 +3415,7 @@ for file in $MALLOC_HEADERS; do
   fi
 done
 
-AC_CHECK_FUNCS(strndup posix_memalign memalign valloc)
+AC_CHECK_FUNCS(strndup posix_memalign memalign)
 
 AC_CHECK_FUNCS(malloc_usable_size)
 
diff --git a/js/src/jsutil.cpp b/js/src/jsutil.cpp
index b7def9f..ab9fca0 100644
--- a/js/src/jsutil.cpp
+++ b/js/src/jsutil.cpp
@@ -69,7 +69,7 @@ AllTheNonBasicVanillaNewAllocations()
         intptr_t(memalign(16, 16)) +
         //intptr_t(q) +
         //intptr_t(aligned_alloc(16, 16)) +
-        intptr_t(valloc(4096)) +
+        //intptr_t(valloc(4096)) +
         intptr_t(strdup("dummy"));
 
     printf("%u\n", uint32_t(p));  // make sure |p| is not optimized away
