From 4572b176be953c639de2a7cbd02cf2415f665b98 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Tue, 28 Oct 2014 16:31:31 +0800
Subject: [PATCH 2/4] Add valloc to pass compile error.

---
 libc/include/malloc.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libc/include/malloc.h b/libc/include/malloc.h
index cb1dd3b..a8dec50 100644
--- a/libc/include/malloc.h
+++ b/libc/include/malloc.h
@@ -36,6 +36,9 @@ extern void free(void* p);
 extern void* memalign(size_t alignment, size_t byte_count) __mallocfunc __wur __attribute__((alloc_size(2)));
 extern size_t malloc_usable_size(const void* p);
 
+extern void* valloc(size_t byte_count) __mallocfunc __wur;
+extern void* pvalloc(size_t byte_count) __mallocfunc __wur;
+
 #ifndef STRUCT_MALLINFO_DECLARED
 #define STRUCT_MALLINFO_DECLARED 1
 struct mallinfo {
-- 
1.9.1

