From 722a24e0abd6586ac5c61641fc150ffbff2f4e89 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Fri, 6 Dec 2013 14:38:10 +0800
Subject: [PATCH] Pass nsAppShell compile error

---
 widget/gonk/nsAppShell.cpp |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/widget/gonk/nsAppShell.cpp b/widget/gonk/nsAppShell.cpp
index c58da87..1805757 100644
--- a/widget/gonk/nsAppShell.cpp
+++ b/widget/gonk/nsAppShell.cpp
@@ -747,10 +747,11 @@ nsAppShell::Init()
     if (XRE_GetProcessType() == GeckoProcessType_Default) {
 #ifdef MOZ_OMX_DECODER
         android::MediaResourceManagerService::instantiate();
-#endif
+// Todo kitkat
 #if ANDROID_VERSION >= 18
         android::FakeSurfaceComposer::instantiate();
 #endif
+#endif
         GonkPermissionService::instantiate();
     }
 
-- 
1.7.9.5

