diff --git a/scx15/init.scx15.rc b/scx15/init.scx15.rc
index 6ccfc7f..0553b69 100755
--- a/scx15/init.scx15.rc
+++ b/scx15/init.scx15.rc
@@ -48,6 +48,11 @@ on fs
     setprop ro.crypto.fuse_sdcard true
     symlink /system/res /res
     symlink /system/bin /bin
+# make b2g binary executable
+    chmod 0755 /system/b2g/b2g
+    chmod 0755 /system/b2g/plugin-container
+    chmod 0755 /system/b2g/updater
+    mount ubifs /dev/ubi0_system /system ro remount
 
 #Allow others to read fb devices
     chmod 660 /dev/graphics/fb0
diff --git a/scx15/nand/fstab.scx15 b/scx15/nand/fstab.scx15
index 6054469..26e76ce 100644
--- a/scx15/nand/fstab.scx15
+++ b/scx15/nand/fstab.scx15
@@ -3,7 +3,7 @@
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
 # <src>                             <mnt_point>     <type>  <mnt_flags and options>                 <fs_mgr_flags>
 
-/dev/ubi0_system                     /system         ubifs  ro,compr=lzo                            wait
+/dev/ubi0_system                     /system         ubifs  rw,compr=lzo                            wait
 /dev/ubi0_userdata                   /data           ubifs  noatime,nosuid,nodev,compr=lzo          wait,encryptable=footer
 /dev/ubi0_cache                      /cache          ubifs  noatime,nosuid,nodev,compr=lzo          wait
 /dev/ubi0_prodnv                     /productinfo    ubifs  noatime,nosuid,nodev,compr=lzo          wait
