From 08f52ffdba35e44adeb2913921f19236c49cca30 Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Thu, 5 Dec 2013 16:05:43 +0800
Subject: [PATCH] Pass compile error

---
 include/ui/PixelFormat.h |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/ui/PixelFormat.h b/include/ui/PixelFormat.h
index 627cfb6..dca0afc 100644
--- a/include/ui/PixelFormat.h
+++ b/include/ui/PixelFormat.h
@@ -63,6 +63,8 @@ enum {
     PIXEL_FORMAT_BGRA_8888   = HAL_PIXEL_FORMAT_BGRA_8888,  // 4x8-bit BGRA
     PIXEL_FORMAT_RGBA_5551   = 6,                           // 16-bit ARGB
     PIXEL_FORMAT_RGBA_4444   = 7,                           // 16-bit ARGB
+    //Todo HAL_PIXEL_FORMAT_sRGB_A_8888
+    PIXEL_FORMAT_A_8         = HAL_PIXEL_FORMAT_sRGB_A_8888,        // 8-bit A
 };
 
 typedef int32_t PixelFormat;
-- 
1.7.9.5

