From e064ca793af2ad70ca3db439fefba2968d75681d Mon Sep 17 00:00:00 2001
From: Kai-Zhen Li <kli@mozilla.com>
Date: Thu, 5 Dec 2013 14:44:08 +0800
Subject: [PATCH 8/8] correct wifi api - wifi_stop_supplicant

---
 dom/wifi/WifiUtils.cpp |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/dom/wifi/WifiUtils.cpp b/dom/wifi/WifiUtils.cpp
index d4a457d..84652d6 100644
--- a/dom/wifi/WifiUtils.cpp
+++ b/dom/wifi/WifiUtils.cpp
@@ -141,7 +141,6 @@ class ICSWpaSupplicantImpl : public WpaSupplicantImpl
 public:
   DEFAULT_IMPL(wifi_load_driver, int32_t, )
   DEFAULT_IMPL(wifi_unload_driver, int32_t, )
-  DEFAULT_IMPL(wifi_stop_supplicant, int32_t, )
 
   DEFINE_DLFUNC(wifi_wait_for_event, int32_t, char*, size_t)
   int32_t do_wifi_wait_for_event(const char *iface, char *buf, size_t len) {
@@ -161,6 +160,12 @@ public:
     return wifi_start_supplicant();
   }
 
+  DEFINE_DLFUNC(wifi_stop_supplicant, int32_t)
+  int32_t do_wifi_stop_supplicant(int32_t) {
+    USE_DLFUNC(wifi_stop_supplicant)
+    return wifi_stop_supplicant();
+  }
+
   DEFINE_DLFUNC(wifi_connect_to_supplicant, int32_t, )
   int32_t do_wifi_connect_to_supplicant(const char* iface) {
     USE_DLFUNC(wifi_connect_to_supplicant)
-- 
1.7.9.5

