diff --git a/src/nci/NfcTagManager.cpp b/src/nci/NfcTagManager.cpp
index 33551de..34c785a 100644
--- a/src/nci/NfcTagManager.cpp
+++ b/src/nci/NfcTagManager.cpp
@@ -756,7 +756,8 @@ bool NfcTagManager::doPresenceCheck()
     return false;
   }
 
-  status = NFA_RwPresenceCheck();
+  //todo
+  //status = NFA_RwPresenceCheck();
   if (status == NFA_STATUS_OK) {
     if (sem_wait(&sPresenceCheckSem)) {
       ALOGE("%s: failed to wait (errno=0x%08x)", __FUNCTION__, errno);
